source/mqtt_request_response.c

```diff
@@ -481,11 +483,198 @@ PyObject *aws_py_mqtt_request_response_client_make_request(PyObject *self, PyObj
+    if (!result) {
+        PyErr_WriteUnraisable(PyErr_Occurred());
+    }
+
+    Py_XDECREF(result);
+
+    PyGILState_Release(state);
+}
+
+PyObject *aws_py_mqtt_request_response_client_create_stream(PyObject *self, PyObject *args) {
+    (void)self;
+
+    PyObject *client_capsule_py;
+    struct aws_byte_cursor subscription_topic_filter;
+    PyObject *subscription_status_changed_callable_py;
+    PyObject *incoming_publish_callable_py;
+
+    if (!PyArg_ParseTuple(
+            args,
+            "Os#OO",
+            /* O */ &client_capsule_py,
+            /* s */ &subscription_topic_filter.ptr,
+            /* # */ &subscription_topic_filter.len,
+            /* O */ &subscription_status_changed_callable_py,
+            /* O */ &incoming_publish_callable_py)) {
+        return NULL;
+    }
+
+    struct mqtt_request_response_client_binding *client_binding =
+        PyCapsule_GetPointer(client_capsule_py, s_capsule_name_mqtt_request_response_client);
+    if (!client_binding) {
+        return NULL;
+    }
+
+    struct mqtt_streaming_operation_binding *stream_binding =
+        s_mqtt_streaming_operation_binding_new(subscription_status_changed_callable_py, incoming_publish_callable_py);
+
+    struct aws_mqtt_streaming_operation_options stream_options = {
+        .topic_filter = subscription_topic_filter,
+        .subscription_status_callback = s_aws_mqtt_streaming_operation_subscription_status_callback_python,
+        .incoming_publish_callback = s_aws_mqtt_streaming_operation_incoming_publish_callback_python,
+        .terminated_callback = s_mqtt_streaming_operation_binding_on_terminated,
+        .user_data = stream_binding,
+    };
+    stream_binding->native =
+        aws_mqtt_request_response_client_create_streaming_operation(client_binding->native, &stream_options);
+    if (stream_binding->native == NULL) {
+        PyErr_SetAwsLastError();
+        s_mqtt_streaming_operation_binding_on_terminated(stream_binding);
+        return NULL;
+    }
+
+    PyObject *capsule = PyCapsule_New(
+        stream_binding, s_capsule_name_mqtt_streaming_operation, s_mqtt_streaming_operation_binding_destructor);
+    if (!capsule) {
+        stream_binding->native = aws_mqtt_rr_client_operation_release(stream_binding->native);
+        return NULL;
+    }
+
+    return capsule;
+}
+
+PyObject *aws_py_mqtt_streaming_operation_open(PyObject *self, PyObject *args) {
+    (void)self;
+
+    PyObject *stream_capsule_py;
+
+    if (!PyArg_ParseTuple(
+            args,
+            "O",
+            /* O */ &stream_capsule_py)) {
+        return NULL;
+    }
+
+    struct mqtt_streaming_operation_binding *stream_binding =
+        PyCapsule_GetPointer(stream_capsule_py, s_capsule_name_mqtt_streaming_operation);
+    if (!stream_binding) {
+        return NULL;
+    }
+
+    if (aws_mqtt_rr_client_operation_activate(stream_binding->native)) {
+        PyErr_SetAwsLastError();
+        return NULL;
+    }
+
+    return Py_None;
```

Comment:
This should use `return Py_RETURN_NONE` instead of `return Py_None`. `Py_None` is ref-counted and its ref-count will be decremented after ` _awscrt.mqtt_streaming_operation_open(self._binding)` is done with the result. `Py_RETURN_NONE` increments the ref of Py_None before returning it. Not incrementing the ref can result in either a deallocating None error or the memory being funky and reused in weird, potentially bad ways.